### ISSUE TITLE
Gradient is None for a parameter in a Bijector in TF 2.0

### ISSUE THREAD

**Issue by esdu** [tensorflow/probability#681]
```
tensorflow 2.0.0
tensorflow_probability 0.8.0
```

I passed a distribution through a simple Bijector with a simple `shift` variable, then I used the gradient tape to capture the gradient of some function w.r.t. the `shift`. The gradient is `[None]`.

(I have a feeling this is not a bug and I just used the library incorrectly... please let me know)

```python
base_dist = tfp.distributions.MultivariateNormalDiag(loc=[0, 0])

shift = tf.Variable([0., 0.], trainable=True)

final_dist = tfp.distributions.TransformedDistribution(
    distribution=base_dist,
    bijector=tfp.bijectors.Affine(shift=shift)
)

with tf.GradientTape() as t:
  loss = -final_dist.log_prob([[10., 10.]])

grads = t.gradient(loss, [shift])

print('loss', loss)
# >> loss tf.Tensor([101.837875], shape=(1,), dtype=float32)

print('grads', grads)
# >> grads [None]
```

Please see this Colab to reproduce: https://colab.research.google.com/drive/1g7HMPmXZ5veRybOWoTUFyg0ZNVn9Ocz7

**Comment by csuter** [comment]
Thanks for reaching out @esdu! We have undertaken a major refactoring effort to make code like what you've written work in TF2+TFP, but we haven't yet shipped a stable release (TFP 0.9) including all the changes. Your code should work great with the tf-nightly and tfp-nightly pip packages, however. Can you give that a try and let us know if it works?

We have one more Distribution to migrate as part of this effort, after which I think we'll be promptly shipping a 0.9 stable release with these changes.

**Comment by jburnim** [comment]
Also:  As part of the refactoring for TF2 that @cgs mentioned, we have deprecated `tfp.bijectors.Affine`, replacing it with new bijectors like `Shift`, `MatvecDiag`, `MatvecTriL`, etc.

When I run your example against `tf-nightly` and `tfp-nightly`, I see gradients printed when I replace `tfp.bijectors.Affine` with `tfp.bijectors.Shift`.

**Comment by esdu** [comment]
Can confirm. Using `tfp.bijectors.Shift` from `tfp-nightly` works great. Looking forward to the 0.9 release. Thanks!
